sdk/identity/identity/src/credentials/azureApplicationCredential.ts

```diff
@@ -8,6 +8,31 @@ import {
 
 import type { AzureApplicationCredentialOptions } from "./azureApplicationCredentialOptions.js";
 import { ChainedTokenCredential } from "./chainedTokenCredential.js";
+import { credentialLogger } from "../util/logging.js";
+import { TokenCredential } from "@azure/core-auth";
+
+const logger = credentialLogger("AzureApplicationCredential");
+
+/**
+ * A no-op credential that logs the reason it was skipped if getToken is called.
+ * @internal
+ */
+export class UnavailableAzureApplicationCredential implements TokenCredential {
```

Comment:
We are discussing this design internally, so we'll hold off making this change for now.